clang/lib/Driver/Driver.cpp

```diff
@@ -2679,7 +2679,7 @@ static SmallVector<const char *, 16> getLinkerArgs(Compilation &C,
         A->getOption().matches(options::OPT_Xlinker)) {
       // Parse through additional linker arguments that are meant to go
       // directly to the linker.
-      std::string PrevArg;
+      static std::string PrevArg;
```

Comment:
Could this change be supplied with a comment that explains its effect, i.e. the scenario where this facilitates the argument retention?
Another option could be to make this a field in the `Driver`/offload builder class.

Reply:
Updated with comment.  It's a pretty local behavior, so I just kept it here.